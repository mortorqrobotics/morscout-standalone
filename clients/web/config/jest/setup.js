global.fetch = require("jest-fetch-mock");

const Chance = require("chance");

const chances = {};

const mockMath = Object.create(Math);
mockMath.random = (seed = 42) => {
  chances[seed] = chances[seed] || new Chance(seed);
  const chance = chances[seed];
  return chance.random();
};

global.Math = mockMath;
