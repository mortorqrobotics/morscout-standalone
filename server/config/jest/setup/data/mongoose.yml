Season:
  - _id: 5d7be5a7994e35048b9e5e0e
    year: 2019
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
  - _id: 4d7be5a7994e35048b9e5e0e
    year: 2015
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
Regional:
  - _id: 6d7be5a7994e35048b9e5e0e
    name: Hi!!!
    key: abc
    url: google.com
    eventType: regional
    startDate: 2019-01-08 12:00:00 -5
    endDate: 2019-02-03 12:00:00 -5
    season: 5d7be5a7994e35048b9e5e0e
    location:
      type: Point
      coordinates: [0,0]
Alliance:
  - _id: 2d7ce5a7994e35048b9e5e0e
    teams:
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
Match:
  - _id: 7d7be5a7994e35048b9e5e0e
    redAlliance: 2d7ce5a7994e35048b9e5e0e
    blueAlliance: 2d7ce5a7994e35048b9e5e0e
    winner: Red
    startDate: 2019-02-03 12:00:00 -5
    endDate: 2019-02-03 12:10:00 -5
    Regional: 6d7be5a7994e35048b9e5e0e
User:
  - _id: 8d7be5a7994e35048b9e5e0f
    team: 9d7be5a7994e35048b9e5e0e
    phone: 1334567890
    email: bo@bob.bob
    firstname: Bob
    lastname: Boby
    password: Secret
    username: bob2
Team:
  - _id: 9d7be5a7994e35048b9e5e0e
    regional: 6d7be5a7994e35048b9e5e0e
    number: 1515
    name: MorTorq
    location:
      type: Point
      coordinates: [0,0]
