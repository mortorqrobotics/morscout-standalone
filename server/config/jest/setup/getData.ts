/* eslint-disable security/detect-object-injection */
import { join } from "path";
import { readFile } from "fs-extra";
import { parse } from "yaml";
import * as models from "../../../src/models";

export default async function getData() {
  let data = {};
  try {
    data = parse(
      await readFile(join(__dirname, "data", "mongoose.yml")).toString()
    );
  } catch (e) {
    console.log(e);
  }
  let promises: Promise<any>[] = [];
  Object.entries(models).forEach(([name, model]) => {
    if (typeof data[name] !== "undefined") {
      data[name].forEach(datum => {
        promises.push((new model(datum) as any).save());
        // (m as any).save().catch(() => {});
      });
    }
  });
  await Promise.all(promises);
}
