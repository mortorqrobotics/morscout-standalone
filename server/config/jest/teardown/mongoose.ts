import mongoose from "mongoose";

module.exports = () => {
  mongoose.disconnect();
  (mongoose as any).server.stop();
};
