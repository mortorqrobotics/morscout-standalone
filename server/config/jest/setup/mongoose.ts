import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";
import getData from "./getData";

let mongoServer: MongoMemoryServer;
export async function setup() {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getConnectionString();
  (process as any).mongoUri = mongoUri;
  await mongoose.connect(
    mongoUri,
    { useNewUrlParser: true, useCreateIndex: true },
    err => {
      if (err) console.error(err);
    }
  );
  await getData();
  return mongoUri;
}

export async function teardown() {
  mongoServer.stop();
}
