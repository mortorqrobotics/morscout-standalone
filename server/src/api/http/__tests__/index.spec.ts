import request from "supertest";
import { setup, teardown } from "setup/mongoose";
import configEmitter from "config";

import router from "../index";

beforeAll(setup);
afterAll(teardown);

test("http has the same routes", async () => {
  expect(router.stack).toMatchSnapshot();
});

test("logs debug message when it gets a route", async () => {
  jest.spyOn(configEmitter.logger, "debug");
  await request(router).get("/");
  expect(configEmitter.logger.debug).toHaveBeenCalled();
});

// test("")
