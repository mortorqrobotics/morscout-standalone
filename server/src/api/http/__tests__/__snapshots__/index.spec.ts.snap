// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http has the same routes 1`] = `
Array [
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "<anonymous>",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\(\\?=\\\\/\\|\\$\\)/i,
    "route": undefined,
  },
  Layer {
    "handle": [Function],
    "keys": Array [
      Object {
        "name": "regional",
        "offset": 10,
        "optional": false,
      },
    ],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/matches\\\\/\\(\\?:\\(\\[\\^\\\\/\\]\\+\\?\\)\\)\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/matches/:regional",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [
      Object {
        "name": "match",
        "offset": 8,
        "optional": false,
      },
    ],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/match\\\\/\\(\\?:\\(\\[\\^\\\\/\\]\\+\\?\\)\\)\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/match/:match",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [
      Object {
        "name": "season",
        "offset": 8,
        "optional": false,
      },
    ],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/teams\\\\/\\(\\?:\\(\\[\\^\\\\/\\]\\+\\?\\)\\)\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/teams/:season",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [
      Object {
        "name": "team",
        "offset": 7,
        "optional": false,
      },
    ],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/team\\\\/\\(\\?:\\(\\[\\^\\\\/\\]\\+\\?\\)\\)\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/team/:team",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/season\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/season",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [
      Object {
        "name": "year",
        "offset": 9,
        "optional": false,
      },
    ],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/season\\\\/\\(\\?:\\(\\[\\^\\\\/\\]\\+\\?\\)\\)\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/season/:year",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
  Layer {
    "handle": [Function],
    "keys": Array [],
    "name": "bound dispatch",
    "params": undefined,
    "path": undefined,
    "regexp": /\\^\\\\/\\?\\$/i,
    "route": Route {
      "methods": Object {
        "get": true,
      },
      "path": "/",
      "stack": Array [
        Layer {
          "handle": [Function],
          "keys": Array [],
          "method": "get",
          "name": "<anonymous>",
          "params": undefined,
          "path": undefined,
          "regexp": /\\^\\\\/\\?\\$/i,
        },
      ],
    },
  },
]
`;
